src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2127,39 +2150,64 @@ var projectUniqueNamesForBuildIntToUpdate
                             var preFetchResult = downloadTasks[nuGetProjectAction.PackageIdentity];
                             using (var downloadPackageResult = await preFetchResult.GetResultAsync())
                             {
+
                                 // use the version exactly as specified in the nuspec file
                                 var packageIdentity = downloadPackageResult.PackageReader.GetIdentity();
-
+                                if(!needsNetstandard20Assets && mightNeedHelperNetstandardPackage && projectTargetFramework != null)
+                                {
+                                    var frameworkReducer = new FrameworkReducer();
+                                    var nearestFramework = frameworkReducer.GetNearest(projectTargetFramework, downloadPackageResult.PackageReader.GetSupportedFrameworks());
+                                    if(nearestFramework == FrameworkConstants.CommonFrameworks.NetStandard20)
+                                    {
+                                        needsNetstandard20Assets = true;
+                                    }
+                                }
                                 await ExecuteInstallAsync(
                                     nuGetProject,
                                     packageIdentity,
                                     downloadPackageResult,
                                     packageWithDirectoriesToBeDeleted,
                                     nuGetProjectContext,
                                     token);
-                            }
-                        }
 
-                        if (nuGetProjectAction.NuGetProjectActionType == NuGetProjectActionType.Install)
-                        {
-                            var identityString = string.Format(CultureInfo.InvariantCulture, "{0} {1}",
+                                var identityString = string.Format(CultureInfo.InvariantCulture, "{0} {1}",
                                 nuGetProjectAction.PackageIdentity.Id,
                                 nuGetProjectAction.PackageIdentity.Version.ToNormalizedString());
 
+                                nuGetProjectContext.Log(
+                                    ProjectManagement.MessageLevel.Info,
+                                    Strings.SuccessfullyInstalled,
+                                    identityString,
+                                    nuGetProject.GetMetadata<string>(NuGetProjectMetadataKeys.Name));
+                            }
+                        }
+                    }
+
+                    if(needsNetstandard20Assets)
+                    {
+
+                        var resolutionContext = new ResolutionContext(DependencyBehavior.Lowest,
+                            includePrelease: true,
```

Comment:
@jainaashish changed the logic to get the latest stable version. if there is no stable version, then get the latest pre-release.